cumulusci/robotframework/Salesforce.py

```diff
@@ -499,6 +525,115 @@ def salesforce_insert(self, obj_name, **kwargs):
         self.store_session_record(obj_name, res["id"])
         return res["id"]
 
+    def _salesforce_generate_object(self, obj_name, **fields):
+        obj = {"attributes": {"type": obj_name}}  # Object type to create
+        obj.update(fields)
+        return obj
+
+    random_string_generator = StringGenerator(String().generate_random_string)
+    faker_template_library = FakerTemplateLibrary()
+
+    def salesforce_collection_generate(self, obj_name, number_to_create, **fields):
+        """Returns an array of dictionaries with template-formatted arguments appropriate for a Collection Insert.
+            Use ``{{number}}`` to represent the unique index of the row in the list of rows, ``{{random_str}}`` to represent a random string
+            and a string that consists of just ``{{number}}`` to generate an actual integer (as opposed to a string-encoded number)
+
+            For example:
+
+                | @{objects} =  Salesforce Collection Generate  Contact  3
+                | ...  FirstName=User {{number}}
+                | ...  LastName={{random_str}}
+                | ...  Age={{number}}
+
+            Which would generate:
+
+                | [{'FirstName': 'User 0', 'LastName': 'u4PNlGUD', 'type': 'Contact', 'Age': '0'},
+                |  {'FirstName': 'User 1', 'LastName': 'Kyd4PC5x', 'type': 'Contact', 'Age': '1'},
+                |  {'FirstName': 'User 2', 'LastName': 'n4EGs4Vp', 'type': 'Contact', 'Age': '2'}]
+
+            Jinja2 Expression Syntax is allowed so computed templates like this are also allowed: {{1000 + number}}
+
+            Templates can also be based on faker patterns like those described here:
+
+            https://faker.readthedocs.io/en/master/providers.html
+
+            Most examples can be pasted into templates verbatim:
+
+        @{objects}=  Salesforce Collection Generate  Contact  200
```

Comment:
This example needs the `|` prefix so that it's treated as preformatted code.

Reply:
Done.